docs/DeveloperGuide.md

```diff
@@ -258,61 +253,35 @@ Key Consideration: How to implement the relationship between `Event` and `Person
-4. A new `Event` is created by `createEditedEvent` method from the details provided by the `EditEventDescriptor`
-5. Lastly the method `setEvent` from model is called which replaces the event at the specified index with the new event containing the edited fields.
-
-The following activity diagram summarizes what happens when a user executes the `EditEventCommand`:
-
-![EditEventActivityDiagram.png](images\EditEvent.png)
-
-#### Design Considerations
-
-- Current implementation 
-  - The current implementation relies on the index for identifying which event is to be edited.
-
-- Alternative implementation Considered
-  - We considered using the `EventName` and `EventDate` to be used to identify the event to be edited. However, we realised that some events can have a very long name thus it is impractical for the user to have to type out the entire event name out, rather using the `showevents` command to identify the index of the event would be much more practical.
-
+* **Current Implementation**
+  * Using a `FilteredList` we are able to segregate events into past events and future events and display these events according to user's preference, this is largely similar to AB3's implementation of `list`, however we need to add the ability to automatically change tabs once this command is called, thus the `CommandResult` class was modified to contain a boolean `isEvent` which is `true` for any command related to events and enables the seamless switching between tabs.
+  * Pros
+    * Allows for easy traversal between the Friends, Events and insights tab.
+    * It is easily extendable, for any new features we simply need to add a boolean which represents that feature, and create an overloaded constructor to initialize that variable to `true`
+    * Enforces the abstraction barrier between the UI component and Logic component
+  * Cons
+    * Although easily extendable, to ensure backward compatibility we need to ensure that the old constructor initialize the boolean for this new feature as `false`
+    * If many features are added, can lead to many overloaded constructor which will make the entire `CommandResult` class convoluted
+
+* **Alternate Implementation**
+  * We can implement a check in `MainWindow` which uses switch cases to detect which command is executed and accordingly switch tabs. The implementation of the representation of `Events` using a `FilteredList` remains the same.
+  * Pros
+    * This is very similar to the implementation in `AddressBookParser`, it easily extendable, as we simply need to add new switch statements for new features and commands
+  * Cons
+    * Violates the Law of Demeter as the UI does not need to know exactly which command is executed, it simply needs to know whether to switch tabs to event or not
```

Comment:
Are you sure it's law of demeter? Sounds more like unnecessary coupling.

Reply:
Yes, my reason is that  `Mainwindow` does not need to know what command has been executed (Whether it is a listevent, or addfriend), it just needs to know whether to switch to the Events tab or the friends Tab.